subprojects/core-api/src/main/java/org/gradle/api/model/ManagedType.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2025 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.model;
+
+import org.gradle.api.Incubating;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+
+/**
+ * Marks types that can be instantiated by Gradle via the managed object infrastructure.
+ * <p>
+ * Types annotated with this annotation will be automatically instantiated by Gradle
+ * when constructing managed objects. See the below example:
+ *
+ * <pre class='autoTested'>
+ * interface MyObject {
```

Comment:
As a plugin author, do I need to mark `MyObject` too? If yes, add it to the javaDoc. 
If not, why is the package not `internal`?

Reply:
You do not need to mark your class it it.

It's not internal since it feels wrong for an annotation on a public type to be internal.

We can add javadoc stating this should only be used by Gradle and it acts as documentation for users